### PR TITLE
CLOUDP-114891: Improve org/project selection when too many orgs/projects

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-114891

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### global user not in VPN
```bash
./bin/mongocli login

First, copy your one-time code: AAAA-AAAA

Next, sign with your browser and enter the code.

Or go to https://account.mongodb.com/account/connect

Your code will expire after 10 minutes.
Successfully logged in as gustavo.bazan@mongodb.com.
Press Enter to continue your profile configuration
There was an error fetching your organizations: Global user is from outside whitelisted subnets.
? Do you want to enter the Org ID manually? Yes
? Default Org ID: [? for help]
```

### More than 100 results
```bash
./bin/mongocli login

First, copy your one-time code: AAAA-AAAA

Next, sign with your browser and enter the code.

Or go to https://account.mongodb.com/account/connect

Your code will expire after 10 minutes.
Successfully logged in as gustavo.bazan@mongodb.com.
Press Enter to continue your profile configuration
You have access to more than 100 orgs
? Do you want to enter the Org ID manually? No
You have access to more than 100 projects
? Do you want to enter the Project ID manually? (y/N)
```

### Less than 100 results
```bash
./bin/mongocli login -P gus

First, copy your one-time code: AAAA-AAAA

Next, sign with your browser and enter the code.

Or go to https://account.mongodb.com/account/connect

Your code will expire after 10 minutes.
Successfully logged in as gssbzn@gmail.com.
Press Enter to continue your profile configuration
? Choose a default organization: Test (z)
? Choose a default project: Test (z)
? Default Output Format:
? Default MongoDB Shell Path: /usr/local/bin/mongosh

Your profile is now configured.
To use this profile, you must set the flag [-P gus] for every command.
You can use [mongocli config set] to change these settings at a later time.
```

### CODE REVIEW

**Comment by boooczek** [comment]
Making some copy suggestions
### global user not in VPN
```
./bin/mongocli login

First, copy your one-time code: AAAA-AAAA

Next, sign with your browser and enter the code.

Or go to https://account.mongodb.com/account/connect

Your code will expire after 10 minutes.
Successfully logged in as gustavo.bazan@mongodb.com.
Press Enter to continue your profile configuration
There was an error fetching your organizations: Global user is outside the Access List entries.
? Enter the default Org ID manually: [? for help]
```

For the below case, would it be possible to do sth like this:
### More than 100 results
```
./bin/mongocli login

First, copy your one-time code: AAAA-AAAA

Next, sign with your browser and enter the code.

Or go to https://account.mongodb.com/account/connect

Your code will expire after 10 minutes.
Successfully logged in as gustavo.bazan@mongodb.com.
Press Enter to continue your profile configuration
? Choose a default organization or enter the Org ID: Test (z)
? Choose a default project or enter the Project ID: Test (z)
```
This way, it could be the same experience for any amount of orgs and projects.

**Comment by dianchenghu** [comment]
I think the proposed solution as a quick fix works well. My only concern is the number "100".  I quickly asked two CLI users how many items in a selection list they think is maximum and they both told me 10, at most 20.   If Jakub could get the number of how many projects in an organization for our users, the data could also be helpful. My suggestion is, if in more than 99% cases, there are less than 20 projects in an Org, then we could go with the solution Guss proposed but change "100" to "20". 

Also some proposed copy changes: 
Press Enter to continue your profile configuration
You have access to more than 20 organizations,
? Instead of selection from the list, do you want to enter the Org ID directly? (y/N)
You have access to more than 100 projects,
? Instead of selection from the list, do you want to enter the Project ID directly? (y/N)

**Review by blva — APPROVED** [review]
LGTM

**Comment by gssbzn** [comment]
@boooczek we can't change the error message here that comes from the API, I can make the change separately from these changes 

@boooczek @dianchenghu sorry if I missed to explain this but...

The results of fetching orgs and projects is paginated with a default page size of 100 and a hard limit of 500 per page, this is where the number 100 came from but feel free to pick any number between 1 and 500 while keeping in mind the following

Currently the org/project selects need to be created with all selectable values before hand, it can't:
a) load new pages as you scroll, nor
b) load new results from the API as you search, nor
c) be both a select and a free form input at the same time, ie allow to manually enter a value if not in the list, a feature request to do this in the lib we use [here](https://github.com/AlecAivazis/survey/issues/339)

all of the above options require investigation time to implement our own custom select and with no clear idea currently if any of those options are really possible 

Based on those points is important for you both to understand that whatever number we decide, ie 20, if the user happens to want to pick the 21st value it will never be in the list nor we would be able to load it and why the current implementation totally skips the list if more than the limit we decide

File: internal/store/logs.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ func (s *Store) Collect(groupID string, newLog *opsmngr.LogCollectionJob) (*opsm
 	}
 }
 
-// ProcessDisks encapsulate the logic to manage different cloud providers.
+// DownloadLog encapsulate the logic to manage different cloud providers.
```
Comment (andreaangiolillo) [inline internal/store/logs.go:81]:
```suggestion
// DownloadLog encapsulates the logic to manage different cloud providers.
```

**Review by andreaangiolillo — DISMISSED** [review]
LGTM

**Comment by boooczek** [comment]
With what @gssbzn said, I'm suggesting to load the max list of 500.
Then we should log a ticket to make an investigation of the desired solution.

**Comment by gssbzn** [comment]
@boooczek I've mede the change to have a limit on 500

**Comment by dianchenghu** [comment]
LGTM

**Review by andreaangiolillo — APPROVED** [review]

**Comment by boooczek** [comment]
Thank you! lgtm

**Pull request merged**